### PR TITLE
Update `liquidatable` `disputeTime` to use `LiquidationTime`

### LINKED ISSUE #895: `disputeTime` needs to be replaced with liquidation time throughout `liquidatable`
right now we record the `uint disputeTime;` variable which should actually be the `liquidation` time. this is the time at which the oracle should be called to dispute against.

### PR DESCRIPTION
**Pull request by chrismaree**
This issue will close #895 which included a bug where `liquidatable` was using the `disputeTime` opposed to the `LiquidationTime` when creating and disputing liquidations.

### CODE REVIEW

File: core/contracts/financial_templates/Liquidatable.sol [hunk 1]
```diff
@@ -184,6 +186,7 @@ contract Liquidatable is PricelessPositionManager {
     function createLiquidation(address sponsor) public returns (uint uuid) {
         // Attempt to retrieve Position data for sponsor
         PositionData storage positionToLiquidate = _getPositionData(sponsor);
+        require(positionToLiquidate.collateral.rawValue > 0, "Cant liquidate a position with no collateral");
```
Comment (nicholaspai) [inline core/contracts/financial_templates/Liquidatable.sol]: why not use `FixedPoint.isGreaterThan(positionToLiquidate.collateral, 0)`?

File: core/contracts/financial_templates/Liquidatable.sol [hunk 2]
```diff
@@ -236,7 +240,7 @@ contract Liquidatable is PricelessPositionManager {
         disputedLiquidation.disputeTime = getCurrentTime();
 
         // Enqueue a request with the DVM.
-        _requestOraclePrice(disputedLiquidation.disputeTime);
+        _requestOraclePrice(disputedLiquidation.liquidationTime);
```
Comment (nicholaspai) [inline core/contracts/financial_templates/Liquidatable.sol]: critical change here, love it

File: core/contracts/financial_templates/Liquidatable.sol [hunk 3]
```diff
@@ -196,6 +199,7 @@ contract Liquidatable is PricelessPositionManager {
                 tokensOutstanding: positionToLiquidate.tokensOutstanding,
                 liquidatedCollateral: positionToLiquidate.collateral.sub(positionToLiquidate.withdrawalRequestAmount),
                 disputer: address(0),
+                liquidationTime: getCurrentTime(),
                 disputeTime: 0,
```
Comment (nicholaspai) [inline core/contracts/financial_templates/Liquidatable.sol]: Do we need `disputeTime` anymore?

File: core/contracts/financial_templates/Liquidatable.sol (hunk 1 above)
Comment (chrismaree) [reply core/contracts/financial_templates/Liquidatable.sol]: ye, that's better. updated

File: core/contracts/financial_templates/Liquidatable.sol [hunk 4]
```diff
@@ -184,6 +186,10 @@ contract Liquidatable is PricelessPositionManager {
     function createLiquidation(address sponsor) public returns (uint uuid) {
         // Attempt to retrieve Position data for sponsor
         PositionData storage positionToLiquidate = _getPositionData(sponsor);
+        require(
+            FixedPoint.isGreaterThan(positionToLiquidate.collateral, 0),
```
Comment (mrice32) [inline core/contracts/financial_templates/Liquidatable.sol]: nit: prefer to use object-like notation for these types of library calls:
```
positionToLiquidate.collateral.isGreaterThan(0)
```

File: core/contracts/financial_templates/Liquidatable.sol [hunk 5]
```diff
@@ -63,7 +63,9 @@ contract Liquidatable is PricelessPositionManager {
 
         // Person who is disputing a liquidation
         address disputer;
-        // Time when dispute is initiated, needed to get price from Oracle
+        // Time when liquidation is initiated, needed to get price from Oracle
+        uint liquidationTime;
+        // Time when the dispute is initialized
```
Comment (mrice32) [inline core/contracts/financial_templates/Liquidatable.sol]: Why can't we get rid of the `disputeTime` variable?

Comment (nicholaspai) [reply core/contracts/financial_templates/Liquidatable.sol]: i had the same comment, I believe we can and should before we merge this

Comment (mrice32) [reply core/contracts/financial_templates/Liquidatable.sol]: ahh, I should've read your comments before I added mine :)

Comment (chrismaree) [reply core/contracts/financial_templates/Liquidatable.sol]: am happy to get rid of it if we are sure there is no need for it on chain. We emit an event on dispute so can always see when the position was disputed anyways.

Comment (chrismaree) [reply core/contracts/financial_templates/Liquidatable.sol]: have removed this.

Comment (mrice32) [reply core/contracts/financial_templates/Liquidatable.sol]: Sweet, thanks.

**Pull request merged**